src/Rings/mpoly-ideals.jl

```diff
@@ -1250,6 +1250,32 @@ julia> codim(I)
 """
 codim(I::MPolyIdeal) = nvars(base_ring(I)) - dim(I)
 
+#######################################################
+#######################################################
+@doc raw"""
+    degree(I::MPolyIdeal)
+
+Return the degree of `I` if it is homogeneous. Otherwise return the degree
```

Comment:
As a non-algebraic geometer: what is the definition of the degree of a (homogeneous) ideal? Do you have a text book reference?

Reply:
One reference I know is  [Michałek and Sturmfels: *Invitation to non-linear algebra*, Section 1.3](https://www.math.uni-konstanz.de/~michalek/NonLinearAlgebra.pdf#page=31). In the homogeneous case, the definition (and the reason for adding this method) are just like [Wolfram says](https://github.com/oscar-system/Oscar.jl/pull/2493#issuecomment-1609401317).

If the ideal is inhomogeneous, Michałek and Sturmfels define the degree to be that of the (homogeneous!) initial ideal with respect to any *degree-compatible* monomial order. Corollary 2.19 in their book then implies that their definition coincides with the degree of the homogenization of the ideal.

Reply:
OK, it's good there is a reference. Still, this is not something that seems to be very general knowledge, so I feel this docstring should at least give a rough description of what "the degree" means in this context. Could you please add something?

Reply:
To be clear: I won't be the one to approve/reject that, I'll leave this to our geometers like @wdecker @afkafkafk13 @simonbrandhorst -- so perhaps we should first hear whether they are willing to merge this, before you waster your valuable time for nothing.

In any case: thank you for your contribution!

Reply:
@taboege Thank you for your contribution. @fingolfin As I already said, a further remark from my side will follow. The code can be made more efficient. I will make a corresponding suggestion, but first have to see a larger picture.

Reply:
@taboege can you add the explanation to the docstring?

Reply:
I added a small paragraph explaining what the degree is, including a reference to the textbook above.